### PR TITLE
[Preview] Separate out functions for react and immutable

### PR DESCRIPTION
**Pull request by bomsy**
Associated Issue: #4829

Here's the Pull Request Doc
https://devtools-html.github.io/debugger.html/CONTRIBUTING.html#pull-requests

### Summary of Changes

* Separate out functions for rendering react and immutable
* remove `size` and `entries`

Todo
* Fix react rendering errors
* Cleanups

Test
* Go to https://accessible-walrus.glitch.me/
* Set breakpoint on line 20
* Refresh page
* When breakpoint gets hit 
* Preview all the possible values above

### Screenshots/Videos (OPTIONAL)
image

### CODE REVIEW

**Comment by jasonLaster** [comment]
it'd be nice to cleanup the actions/preview stuff

**Comment by bomsy** [comment]
> it'd be nice to cleanup the actions/preview stuff

Yeah totally! Will

**Comment by jasonLaster** [comment]
That screenshot is beautiful!

**Comment by jasonLaster** [comment]
do you think we can use `result` to determine whether the object is react or immutable before doing the expensive `client.evaluate` calls?

Anything that would speed us up would be good. We're really slow now :)

**Comment by bomsy** [comment]
Yes! looking into it.

File: src/actions/preview.js [hunk 1]
```diff
@@ -20,14 +20,54 @@ import { isEqual } from "lodash";
...
+  return !!(_root && __ownerID && __altered && __hash);
+}
+
+async function getExtraProps(client, frame, expression, result) {
+  let immutableType = null;
+  let immutableEntries = null;
+  const isImmutable = isImmutableResult(result.preview);
+  const frameId = frame.id;
+
+  const reactDisplayName = await client.evaluate(
```
Comment (jasonLaster) [inline src/actions/preview.js]: can we do the same thing and check to see if it `isReact` as well before client.evaluate

**Pull request merged**